core/services/user_service/middleware/authenticate_token.py

```diff
@@ -0,0 +1,26 @@
+# middleware/authenticate_token.py
+from fastapi import Request, HTTPException
+from starlette.middleware.base import BaseHTTPMiddleware
+from utils.helper.jwt import create_token, verify_token
+
+
+class JWTAuthMiddleware(BaseHTTPMiddleware):
+    def __init__(self, app, secret_key: str):
+        super().__init__(app)
+        self.secret_key = secret_key
+
+    async def dispatch(self, request: Request, call_next):
+        auth_header = request.headers.get("Authorization")
+        if auth_header:
+            token = auth_header.split(" ")[1]
```

Comment:
What happens if the header contains no whitespace?

Reply:
It will raise the HTTPException, do you have a better way to substitude the .split() I didn't find anything better